src/backend/vulkan/src/window.rs

```diff
@@ -286,13 +348,18 @@ impl Instance {
             instance: self.raw.clone(),
         });
 
-        Surface { raw, width, height, samples }
+        Surface {
+            raw,
+            current_extent: None,
+            samples,
+            connection: Arc::new(Mutex::new(connection)),
+        }
     }
 }
 
 impl hal::Surface<Backend> for Surface {
     fn kind(&self) -> hal::image::Kind {
-        hal::image::Kind::D2(self.width, self.height, 1, self.samples)
+        hal::image::Kind::D2(self.current_extent.unwrap().width, self.current_extent.unwrap().height, 1, self.samples)
```

Comment:
I feel like most of the effort went into implementing this function but I don't believe we actually need it.

Reply:
We need to update the extent for swapchain recreation (so the swapchain images are the right size). Caching it for the kind function adds like ~10 lines.

Reply:
Correct me if I'm wrong why do we need to cache those at all? Users should just call createswapchain with the old swapchain and the new size gathered from the resize event of the native window or winit. For additional safety checks usere can requery the surface capabilties the check the current extent (optional per platform) and check if the extents are inside the boundaries. Updating these are done by the vulkan drivers already.